surefire-providers/surefire-testng/src/main/java/org/apache/maven/surefire/testng/conf/TestNG740Configurator.java

```diff
@@ -37,33 +36,77 @@
  *
  * @since 3.0.0-M6
  */
-public class TestNG740Configurator extends TestNG60Configurator
+public class TestNG740Configurator
+    extends TestNG60Configurator
 {
+    /**
+     * Convert and apply the value of the [threadcount] option.
+     * 
+     * @param suite TestNG {@link XmlSuite} object
+     * @param options Surefire plugin configuration options
+     * @throws TestSetFailedException if unable to convert specified [threadcount] setting
+     */
     @Override
-    public void configure( XmlSuite suite, Map<String, String> options )
+    protected void configureThreadCount( XmlSuite suite, Map<String, String> options )
         throws TestSetFailedException
     {
-        String threadCountAsString = options.get( THREADCOUNT_PROP );
-        int threadCount = threadCountAsString == null ? 1 : parseInt( threadCountAsString );
-        suite.setThreadCount( threadCount );
-
-        String parallel = options.get( PARALLEL_PROP );
-        if ( parallel != null )
+        String threadCount = options.get( THREADCOUNT_PROP );
+        // if [threadcount] spec'd
+        if ( threadCount != null )
         {
-            if ( !"methods".equalsIgnoreCase( parallel ) && !"classes".equalsIgnoreCase( parallel ) )
+            try
             {
-                throw new TestSetFailedException( "Unsupported TestNG parallel setting: "
-                    + parallel + " ( only METHODS or CLASSES supported )" );
+                // convert and apply [threadcount] setting
+                suite.setThreadCount( Integer.parseInt( threadCount ) );
```

Comment:
Previously it was `threadCountAsString == null ? 1 : ...`. What would happen if parallel=classes but the ThreadCount is not set? How is TestNG implemented? It has default value 1?

Reply:
I found the default value in TestNG:
`private int m_threadCount = 5;`
We should use the same as before, means `suite.setThreadCount( threadCountAsString == null ? 1 : Integer.parseInt( threadCount ) )`

Reply:
If all supported versions of TestNG provide a default for this setting, it may be best to not inject a default here. Just let TestNG provide its own default value. That's how the new code for TestNG 7.4+ behaves. I don't know the history of the original code to understand why this was coded to inject a default value of 1.

Reply:
Why would we override the default value provided by TestNG? What's the benefit of specifying parallel execution if the thread count is 1?

Reply:
This PR should not change the behavior. It should fix the value `parallel=none` and other previously unknown values for the same parameter.

Reply:
I can remove the implementation that bypasses the override of TestNG's default thread count, but it's unclear why the original code behaves this way. As it's currently implemented, specifying any form of parallel execution without explicitly specifying thread count will result in sequential execution. 
The documentation [here](https://maven.apache.org/surefire/maven-surefire-plugin/examples/testng.html#Running_Tests_in_Parallel) indicates that the default is 5, which would be true if the implementation in Surefire didn't impose a default value of 1.

Reply:
We should not change behavior in this PR. We are only aiming for the fix regarding configuration with the parameter `parallel`.
Speaking apart of thie PR, I would say that `5` is a kind of Bulgarian value in IT. The other frameworks do not use arbitrary values like TestNG does. The surefire providers should behave similar and so the `surefire-junit47` has fallback value `0`. The developer who wrote the documentation obviously had a problem with `5`. Not sure if TestNG would have a problem with `0` (main Thread) and if `1` is a singleton non-main Thread. Feel free to check it out.

Reply:
@Tibor17 As you indicated, TestNG provides a default value of 5. As I pointed out, the Surefire documentation also specifies 5 as the default value. There is no potential for the thread count to have anything other than 5 unless an explicit assignment is made.
I'll remove the code that overrides the default implementation for the thread count parameter. I'll open another issue regarding the improper behavior of the existing implementation (forcing the thread count to 1 in the absence of an explicit assignment).

Reply:
New issue opened here: https://issues.apache.org/jira/browse/SUREFIRE-2075